test/functional/output_kafka.go

```diff
@@ -0,0 +1,153 @@
+	if err := f.Test.Client.Create(brokersecret); err != nil {
+		return err
+	}
+
+	//standup pod with container running broker
+	b.AddInitContainer("init-config1", ImageRemoteKafkaInit).
+		AddEnvVarToInitContainer("NODE_NAME", "functional-test-node", 1).
+		AddEnvVarToInitContainer("POD_NAME", "functional", 1).
+		AddEnvVarToInitContainer("POD_NAMESPACE", b.Pod.Namespace, 1).
+		AddEnvVarToInitContainer("ADVERTISE_ADDR", fmt.Sprintf("%s.%s.svc.cluster.local", kafka.DeploymentName, b.Pod.Namespace), 1).
+		WithCmdArgsToInitContainer([]string{"/bin/bash", "/etc/kafka-configmap/init.sh"}, 1).
+		AddVolumeMountToInitContainer("brokerconfig", "/etc/kafka-configmap", "", false, 1).
+		AddVolumeMountToInitContainer("configkafka", "/etc/kafka", "", false, 1).
+		AddVolumeMountToInitContainer("extensions", "/opt/kafka/libs/extensions", "", false, 1)
+
+	cmdCreateTopicAndDeployBroker := "./bin/kafka-server-start.sh /etc/kafka-configmap/server.properties"
+
+	log.V(2).Info("Adding container", "name", name)
+	b.AddContainer(kafkaBrokerContainerName, ImageRemoteKafka).
+		AddEnvVar("CLASSPATH", "/opt/kafka/libs/extensions/*").
+		AddEnvVar("KAFKA_LOG4J_OPTS", "-Dlog4j.configuration=file:/etc/kafka/log4j.properties").
+		AddEnvVar("JMX_PORT", strconv.Itoa(int(kafkaJMXPort))).
+		AddContainerPort("inside", kafkaInsidePort).
+		AddContainerPort("outside", kafkaOutsidePort).
+		AddContainerPort("jmx", kafkaJMXPort).
+		//WithCmd("./bin/kafka-server-start.sh /etc/kafka-configmap/server.properties").
+		WithCmdStringSlice([]string{"/bin/bash", "-c", cmdCreateTopicAndDeployBroker}).
+		AddVolumeMount("brokerconfig", "/etc/kafka-configmap", "", false).
+		AddVolumeMount("configkafka", "/etc/kafka", "", false).
+		AddVolumeMount("brokerlogs", "/opt/kafka/logs", "", false).
+		AddVolumeMount("brokercert", "/etc/kafka-certs", "", false).
+		AddEnvVar("extensions", "/opt/kafka/libs/extensions").
+		AddEnvVar("data", "/var/lib/kafka/data").
+		End().
+		AddConfigMapVolume("brokerconfig", kafka.DeploymentName).
+		AddSecretVolume("brokercerts", kafka.DeploymentName).
+		AddEmptyDirVolume("brokerlogs").
+		AddEmptyDirVolume("extensions")
+
+	/////////////////////////////////////////////
+	//step c
+	topics := []string{kafka.AppLogsTopic}
+
+	for _, topic := range topics {
+		containername := kafka.ConsumerNameForTopic(topic)
+		// Deploy consumer
+		//consumerapp := kafka.NewKafkaConsumerDeployment(OpenshiftLoggingNS, topic)
+
+		log.V(2).Info("Creating Broker Consumer app")
+		///create topic using --bootstrap-server localhost:9093
+		cmdCreateTopic := fmt.Sprintf(`./bin/kafka-topics.sh --zookeeper localhost:2181 --create --if-not-exists --topic %s --partitions 1 --replication-factor 1 ;`, topic)
+		cmdRunConsumer := fmt.Sprintf(`./bin/kafka-console-consumer.sh --bootstrap-server %s --topic %s --from-beginning > /shared/consumed.logs ;`, "localhost:9092", topic)
+		cmdSlice := []string{"sleep 120;", cmdCreateTopic, cmdRunConsumer}
```

Comment:
Do we really need to wait two minutes ... this adds a lot of time to the test ... can this be polling/wait for something stage somehow and not sleep. The idea of functional tests are to be quick ... so 120 seconds might be a lot of time here.